addons/website_livechat/static/src/models/visitor/visitor.js

```diff
@@ -0,0 +1,133 @@
+            if ('history' in data) {
+                data2.history = data.history;
+            }
+            if ('is_connected' in data) {
+                data2.is_connected = data.is_connected;
+            }
+            if ('lang' in data) {
+                data2.lang = data.lang;
+            }
+            if ('name' in data) {
+                data2.name = data.name;
+            }
+            if ('partner_id' in data) {
+                if (data.partner_id) {
+                    data2.partner = [['insert', { id: data.partner_id }]];
+                } else {
+                    data2.partner = [['unlink-all']];
+                }
+            }
+            if ('website' in data) {
+                data2.website = data.website;
+            }
+            return data2;
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeAvatarSrc() {
+            if (!this.partner) {
+                return '/mail/static/src/img/smiley/avatar.jpg';
+            }
+            return `/web/image/res.partner/${this.partner.id}/image_128`;
+        }
+
+        /**
+         * @private
+         * @returns {mail.country}
+         */
+        _computeCountry() {
+            if (this.partner && this.partner.country) {
+                return [['link', this.partner.country]];
+            }
+            if (this.country) {
+                return [['link', this.country]];
+            }
+            return [['unlink-all']];
+        }
+
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeNameOrDisplayName() {
+            if (this.partner) {
+                return this.partner.nameOrDisplayName;
+            }
+            return this.name;
+        }
+    }
+
+    Visitor.fields = {
+        avatarSrc: attr({
+            compute: '_computeAvatarSrc',
+            dependencies: [
+                'partner',
+            ]
+        }),
+        country: many2one('mail.country', {
+            compute: '_computeCountry',
+            dependencies: [
+                'partnerCountry',
+            ]
+        }),
+        history: attr(),
+        is_connected: attr(),
+        lang: attr(),
+        name: attr(),
+        nameOrDisplayName: attr({
+            compute: '_computeNameOrDisplayName',
+            dependencies: [
+                'name',
+                'partnerNameOrDisplayName',
+            ],
+        }),
+        partner: many2one('mail.partner'),
+        partnerCountry: many2one('mail.country',{
+            related: 'partner.country',
+        }),
+        partnerNameOrDisplayName: attr({related: 'partner.nameOrDisplayName'}),
+        threads: one2many('mail.thread', {
+            inverse: 'visitor',
+        }),
+        website: attr(),
```

Comment:
Ideally all fields here should be documented, but I'm requesting it especially for this one because I have no idea what it contains from looking here. By default I would expect it to contain a `website` record, but that's not the case.